Save load application preferences

Tukop opened the pull request.

Commit:
added snapgrid element to the project

Commit:
binded toolbar grid spinner to the snapgrid

Commit:
added grid view functionnalit

Commit:
now when updating the spacing spinner, the grid is updated

Commit:
now create shape snaps the new shape to the grid. **grid points and grid lines are not sync correctly**

Commit:
fix snapping mode on lastclickposition not being snap to the grid

Tukop committed:
Added application preference save/load

The data saved isn't correct yet

Tukop committed:
Added application preference save/load

The data saved isn't correct yet

Tukop committed:
Added application preference save/load

The data saved isn't correct yet

Tukop committed:
Added application preference save/load

The data saved isn't correct yet

Tukop committed:
Added the strokeSize, strokeColor and fillColor to the preferences

Tukop committed:
Added the strokeSize, strokeColor and fillColor to the preferences

Tukop committed:
Added the strokeSize, strokeColor and fillColor to the preferences

Tukop committed:
Added the strokeSize, strokeColor and fillColor to the preferences

boscoa committed:
Added composite pattern to select multiple shapes.

Tukop committed:
Added the lastLoad and lastSave repository preferences

Also added protection against empty text file when first launching

Tukop committed:
Added the lastLoad and lastSave repository preferences

Also added protection against empty text file when first launching

Tukop committed:
Added the lastLoad and lastSave repository preferences

Also added protection against empty text file when first launching

Tukop committed:
Added the lastLoad and lastSave repository preferences

Also added protection against empty text file when first launching

Commit:
fix snapgrid canvas out-of-bounds

Commit:
remove debugger for grid

Commit:
made some changes to reduce the McCabe Cyclomatic Complexity

pgarneau commented on config.properties:
```diff
@@ -0,0 +1,6 @@
+#Sat Feb 09 18:36:29 EST 2019
```
Put this file inside the .gitignore and remove from commit

pgarneau commented on src/main/java/apaintus/Main.java:
```diff
@@ -1,58 +1,143 @@
+				primaryStage.setHeight(Double.valueOf(properties.getProperty("primaryStageHeight")));
+				primaryStage.setWidth(Double.valueOf(properties.getProperty("primaryStageWidth")));
+			}
+			catch(Exception ex) {
```
Je devine que t'etais oblige de mettre ca dans un try/catch? 

Catcher "Exception" c'est generally bad

pspino replied:
en theorie, si je me trompe pas cela devrait etre un IOException: https://docs.oracle.com/javase/7/docs/api/java/io/IOException.html

Tukop replied:
le try catch est gone

pgarneau commented on src/main/java/apaintus/controllers/MenuController.java:
```diff
@@ -109,5 +113,26 @@ public boolean hasUnsavedWork() {
+	    	pngFileService.setLastSaveRepository(Paths.get(properties.getProperty("lastSaveRepository")));
+	    	xmlFileService.setLastSaveRepository(Paths.get(properties.getProperty("lastSaveRepository")));
+    	} 
+    	catch(Exception ex) {
```
Same comment as above pour "Exception ex"

pgarneau commented on src/main/java/apaintus/controllers/ToolBarController.java:
```diff
@@ -128,4 +141,31 @@ private void setColorPicker(ColorPicker colorPicker, String colorValue) {
+
+    	try {
+	    	strokeSize.getValueFactory().setValue(Double.valueOf(properties.getProperty("strokeSize")));
+		} catch (Exception ex) {
```
same things here

boscoa replied:
Une raison pourquoi sur plusieurs try/catch? Il n'est pas possible de les imbriquer dans un seul?

Tukop committed:
Added config.properties

Tukop committed:
Added config.properties

Tukop committed:
Added config.properties

Tukop committed:
Added config.properties

boscoa committed:
Added ungroup action. Moved vertices for loop inside selection box.

Ungroup button added to attributes panel. This will trigger an ActionEvent that clears the selectionBox and remove it from drawnShapes.

Commit:
added review suggested changes

Commit:
changes from review

Commit:
rename changed snapgrid package name as it was wrongly written

Commit:
white spaces

Commit:
moved activateSnapGrid to upper lines to avoid confusion of it to be living inside private classes of CanvasController

Commit:
removed useless constructor

Commit:
shit shat shot

Tukop committed:
Created a Preferences Service

Modified the code to abide by the new structure

Tukop committed:
Created a Preferences Service

Modified the code to abide by the new structure

Tukop committed:
Created a Preferences Service

Modified the code to abide by the new structure

Tukop committed:
Created a Preferences Service

Modified the code to abide by the new structure

Tukop committed:
removed unused imports

Tukop committed:
removed unused imports

Tukop committed:
removed unused imports

Tukop committed:
removed unused imports

pgarneau committed:
Merge pull request #9 from pgarneau/snapgrid

Snapgrid

Tukop committed:
Merge remote-tracking branch 'origin/Save_load_application_preferences' into Save_load_application_preferences

boscoa committed:
Added translation and rotation to composite shapes

Commit:
fix a line that had x3 the same args in it instead of the right ones

boscoa committed:
Rebase onto master

boscoa committed:
Updated SelectionBox

Calls boundingBox update instead of update, preventing a new update on fields.
Removed update method, it was the same as DrawableShape

boscoa committed:
Merge pull request #10 from pgarneau/multiple_selection

Multiple selection

Tukop committed:
Changed the Application Preference Architecture

Removed lots of spaces

Tukop committed:
Changed the Application Preference Architecture

Removed lots of spaces

Tukop committed:
Merge remote-tracking branch 'origin/Save_load_application_preferences' into Save_load_application_preferences

Tukop committed:
Prevent crash on first launch
